src/strands/telemetry/tracer.py

```diff
@@ -566,7 +585,11 @@ def end_event_loop_cycle_span(
                 )
             else:
                 self._add_event(span, "gen_ai.choice", event_attributes=event_attributes)
-        self._end_span(span, attributes, error)
+
+        # Note: self._end_span() is commented out because when using trace_api.use_span(end_on_exit=True),
```

Comment:
Callers have no control over this; we can leave an explaining comment saying "We don't use self._end_span() because callers are responsible via closing"

But don't leave this commented out, just remove it